app/components/Article/Contents.tsx

```diff
@@ -40,6 +45,7 @@ const addPopup = (e: HTMLElement, id: string, contents: string, mobile?: boolean
     const toggle = () => popup.classList.toggle('shown')
     popup.addEventListener('click', togglePopup(toggle, e))
     e.addEventListener('click', togglePopup(toggle, e))
+    popup.children[0].addEventListener('click', (e) => e.stopPropagation())
   }
 
   return popup
```

Comment:
@mruwnik, ok so there is a problem with multiple popups being added to a reference item for definitions. I think specifically, because the id is generated randomly, there can still exist two popups associated with the same reference whenever the function which creates the random ids is run more than once. And because react tends to just run things randomly...

Reply:
I tried fixing this myself but I couldn't figure it out

Reply:
hmm. You're right :/ One way round this would be to pass in the id, rather than use a random one. So normal articles would just leave it empty, and chatbot ones would pass in their index in the history. 

Otherwise the id would have to be generated based on the contents somehow, which would work for chat responses (as most of the time they will be unique), but not for articles in the chatbot (unless they have some magic token added secretly, which is also ugly). In this case the randomId would just be a hash of the `html` argument or something

Reply:
I've tried again but I think this problem has far deeper and darker roots...
<img width="607" alt="image" src="https://github.com/StampyAI/stampy-ui/assets/148247390/2b7f07f6-0100-4b3b-9fcb-c2855d2bf4ea">
They are replicating inside each-other. Its like trying to fight hydra